Java/Operating-System/banker.java

```diff
@@ -0,0 +1,311 @@
+import java.util.Scanner;
+
+class algorithm {
+
+    void display(int r, int p, int[] avail, int max[][], int need[][], int allocation[][], int res[]) {
+        System.out.println("THE MAXIMUM RESOURCES MATRIX IS AS FOLLOWS :");
+        System.out.print("   ");
+        for (int i = 0; i < r; i++) {
+            System.out.print(" " + (i + 1));
+        }
+        System.out.println(" ");
+        System.out.println("_______________________");
+        for (int i = 0; i < p; i++) {
+            System.out.print("P" + i + "|");
+            for (int j = 0; j < r; j++) {
+                System.out.print(" " + max[i][j]);
+            }
+            System.out.println(" ");
+        }
+
+        System.out.println(" ");
+
+        System.out.println("THE ALLOCATION MATRIX IS AS FOLLOWS :");
+        System.out.print("   ");
+        for (int i = 0; i < r; i++) {
+            System.out.print(" " + (i + 1));
+        }
+        System.out.println(" ");
+        System.out.println("_______________________");
+        for (int i = 0; i < p; i++) {
+            System.out.print("P" + i + "|");
+            for (int j = 0; j < r; j++) {
+                System.out.print(" " + allocation[i][j]);
+            }
+            System.out.println(" ");
+        }
+        System.out.println("");
+
+
+        System.out.print("AVAILABLE ARRAY IS :");
+        for (int i = 0; i < r; i++) {
+            int sum = 0;
+            avail[i] = 0;
+            for (int j = 0; j < p; j++) {
+                sum = sum + allocation[j][i];
+                avail[i] = res[i] - allocation[j][i];
```

Comment:
@shraiyya You can add a comment here about `Checking for available resources inline`.